app/assets/javascripts/discourse/components/site-header.js.es6

```diff
@@ -74,6 +209,14 @@ const SiteHeaderComponent = MountWidget.extend(Docking, {
         this.eventDispatched("dom:clean", "header");
       }
     });
+
+    if (this.site.mobileView) {
```

Comment:
Would it be better to check the capabilities object for touch? Some touch devices like ipad are not in mobileView. `this.capabilities.touch`